### ISSUE TITLE
accessibilityRole prop on Button component can not take different values

### ISSUE THREAD

**Issue by YajurG** [microsoft/react-native-windows#12176]
### Problem Description

Setting the value of the `accessibilityRole` property on a Button component to something other than 'button' does not change the value of the property, as the value of the Control Type in the UIA tree always remains as 'button', as does the value of the `accessibilityRole` property in the debugger. This issue repro's on both Paper and Fabric. Is this issue expected behavior, as the expectation is that the Button component's accessibilityRole value will usually be 'button', but wondering if there should be an option to set the value to the desired role, and default to 'button' with no value specified.

Example (with `accessibilityRole='link'`):
![image](https://github.com/microsoft/react-native-windows/assets/30995198/164270fd-ff87-4518-8b61-c1232fe2cf10)

### Steps To Reproduce

1. Launch playground
2. Navigate to the Button page in RNTester
3. Open Accessibility Insights for Windows
4. Navigate to a Button component
5. Check the 'Control Type' value in the UIA properties section

### Expected Results

Expected the UIA Control Type an LocalizedControlType to match the `accessibilityRole` value and be 'link'.

### CLI version

12.0.0-alpha.7

### Environment

```markdown
System:
  OS: Windows 10 10.0.22621
  CPU: "(24) x64 AMD Ryzen Threadripper PRO 3945WX 12-Cores     "
  Memory: 45.51 GB / 63.86 GB
Binaries:
  Node:
    version: 18.12.1
    path: C:\Program Files\nodejs\node.EXE
  Yarn:
    version: 1.22.19
    path: C:\Program Files (x86)\Yarn\bin\yarn.CMD
  npm:
    version: 8.19.2
    path: C:\Program Files\nodejs\npm.CMD
  Watchman: Not Found
SDKs:
  Android SDK: Not Found
  Windows SDK:
    AllowDevelopmentWithoutDevLicense: Enabled
    AllowAllTrustedApps: Enabled
    Versions:
      - 10.0.18362.0
      - 10.0.19041.0
      - 10.0.22000.0
      - 10.0.22621.0
IDEs:
  Android Studio: AI-213.7172.25.2113.9123335
  Visual Studio:
    - 17.7.34031.279 (Visual Studio Enterprise 2022)
    - 16.11.34031.81 (Visual Studio Enterprise 2019)
Languages:
  Java:
    version: 18.0.2.1
    path: C:\Program Files\Common Files\Oracle\Java\javapath\javac.EXE
  Ruby: Not Found
npmPackages:
  "@react-native-community/cli": Not Found
  react: Not Found
  react-native: Not Found
  react-native-windows: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: Not found
  newArchEnabled: Not found
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found
```

### Target Platform Version

None

### Visual Studio Version

Visual Studio 2022

### Build Configuration

Debug

**Comment by chrisglein** [comment]
@YajurG Can you add a link to the RN core source where it's shown to be not possible to customize this?
And if that's a won't fix for RN core then we'd leave it as a won't fix for this (but this issue in closed state captures the gap)

**Comment by YajurG** [comment]
I think this is the [source](https://github.com/facebook/react-native/blob/b27fab70c52e613982f07a6a79699ff23afd4e59/packages/react-native/Libraries/Components/Button.js#L370). It looks like accessibilityRole is always set to "button" instead of whatever `accessibilityRole` is set to.

**Comment by YajurG** [comment]
> @YajurG Can you add a link to the RN core source where it's shown to be not possible to customize this? And if that's a won't fix for RN core then we'd leave it as a won't fix for this (but this issue in closed state captures the gap)

Should we classify this as 'won't fix' and then close issue accordingly?
